### PR TITLE
CI updates

### PR DESCRIPTION
**Pull request by TriplEight**
- `Swatinem/rust-cache` handles Rust cache better

### CODE REVIEW

**Comment by kpp** [comment]
There is no need to fix CI, it works: https://github.com/libp2p/rust-libp2p/pull/2227/checks

File: .github/workflows/ci.yml
Comment (thomaseizinger) [inline .github/workflows/ci.yml:58]: Seems like an unnecessary diff.

File: .github/workflows/ci.yml
Comment (thomaseizinger) [inline .github/workflows/ci.yml:72]: Same as with the `checkout` action, specifying the full version will avoid a direct dependabot PR after this is merged.

**Review by thomaseizinger — CHANGES_REQUESTED** [review]
A few nits but otherwise a welcoming change from my end. Personally I've had a good experience with `rust-cache` so far!

File: .github/workflows/ci.yml
Comment (TriplEight) [reply .github/workflows/ci.yml:76]: yeah, it's not needed anymore.

File: .github/workflows/ci.yml
Comment (TriplEight) [reply .github/workflows/ci.yml]: you're right

**Review by thomaseizinger — APPROVED** [review]
Thanks, this looks good!

We will have to wait for @mxinden to merge this. He is currently on vacation though so that might take a little longer :)

**Review by mxinden — APPROVED** [review]
Thank you @TriplEight. Very much appreciate your help here! Good to have a GitHub CI expert around.

**Pull request merged**
